src/pb_stub.cc

```diff
@@ -1241,6 +1256,7 @@ main(int argc, char** argv)
             // messages to the backend ASAP.
             if (stub->LogServiceActive()) {
               stub->TerminateLogRequestThread();
+              logger->SetBackendLoggingActive(false);
```

Comment:
Can this be part of `TerminateLogRequestThread`?

Reply:
Part of the reason the reason this seg fault existed is that the Logger and Stub objects were interleaved and were being instantiated at unintended times. Normally, the way we handle it would be fine, but the GIL lock makes this process a bit sensitive. Calling the stub in the internal Log() method is unavoidable, but this was a conscious effort to keep these classes as independent as possible. I can quickly add these methods of the launch/terminate thread methods, however, and you can choose which you prefer.

Reply:
I'm just saying would it be possible to make `Logger` completely part of the `Stub` process to make sure that it doesn't outlive the Stub object which may lead to these issues.

Reply:
Yes, that should be possible. Though it may have to be "marked for improvement", as I think this PR should make into the release if the logging API is going to be released during this cycle.